packages/contracts/contracts/plugin/PluginRepo.sol

```diff
@@ -40,21 +42,23 @@ contract PluginRepo is
     /// @notice The ID of the permission required to call the `createVersion` function.
     bytes32 public constant UPGRADE_REPO_PERMISSION_ID = keccak256("UPGRADE_REPO_PERMISSION");
 
-    /// @notice The index of the next version to be created.
-    uint256 internal nextVersionIndex;
+    /// @notice Current latest release id(there can be maximum uint8 = 255 possibility).
+    uint256 public latestReleaseId;
 
-    /// @notice The mapping between version indices and version information.
-    mapping(uint256 => Version) internal versions;
+    /// @notice increasing build ids per release
+    /// @dev keys can only be uint8 even though uint256 is specified to reduce gas cost.
+    mapping(uint256 => uint256) internal buildIdsPerRelease;
 
-    /// @notice A mapping between the semantic version number hash and the version index.
-    mapping(bytes32 => uint256) internal versionIndexForSemantic;
+    /// @notice The mapping between version hash and its corresponding Version information.
+    /// @dev key: keccak(abi.encode(releaseId, buildId)) value: Version struct.
+    mapping(bytes32 => Version) internal versions;
```

Comment:
do we need a version hash here? wouldn't it be easier to have a 2-level awway that would be iterated like `versions[release][build]`?

Maybe it's more efficient to just hash releaseId+buildId and index using subgraph :+1: